lib/adapter.js

```diff
@@ -4866,6 +4927,25 @@ function Adapter(options) {
         });
     };
 
+    /**
+     * This method update the cached values in this.usernames
+     *
+     * @returns {Promise<void>}
+     */
+    const updateUsernameCache = async () => {
+        // make sure cache is cleared
+        this.usernames = {};
```

Comment:
This can produice  edge cases or !? Because you clear cache and then start an async process to fill again. So this can produce multiple updates to be started "at the same time". But these are real edge caes I think ... But still to have the action "atomic" I would move the "set empty" into the async result method/callback. Yes then still stuff happens in parallel - but only if cache was empty

Reply:
Good point